### PR TITLE
PROTOCOLS-116 Add MPT test for UID<-> MSN handling by IMAP

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[Building Java 8 protocols-116-5f902969-5abd-4362-9a5e-d0c1b85e343a] Tests in error: 
[Building Java 8 protocols-116-5f902969-5abd-4362-9a5e-d0c1b85e343a]   Unauthorized
[Building Java 8 protocols-116-5f902969-5abd-4362-9a5e-d0c1b85e343a]   Unauthorized
```

Unrelated to this code change ;-)

**Comment by chibenwa** [comment]
I'm not in a lucky mood today : 

```
[Building Java 8 protocols-116-62961f73-79ed-4df7-9bdf-69b7e5b4384a] Results :
[Building Java 8 protocols-116-62961f73-79ed-4df7-9bdf-69b7e5b4384a] 
[Building Java 8 protocols-116-62961f73-79ed-4df7-9bdf-69b7e5b4384a] Tests in error: 
[Building Java 8 protocols-116-62961f73-79ed-4df7-9bdf-69b7e5b4384a]   CassandraSieveRepositoryTest>AbstractSieveRepositoryTest.renameScriptShouldPropagateActiveScript:257 » ScriptNotFound
[Building Java 8 protocols-116-62961f73-79ed-4df7-9bdf-69b7e5b4384a] 
[Building Java 8 protocols-116-62961f73-79ed-4df7-9bdf-69b7e5b4384a] Tests run: 266, Failures: 0, Errors: 1, Skipped: 0
```

**Comment by chibenwa** [comment]
test this please

File: mpt/impl/imap-mailbox/core/src/main/resources/org/apache/james/imap/scripts/ConcurrentAppendUIDToMSNMapping.test [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+C: 1e APPEND expungeresponse {254+}
+C: Date: Mon, 7 Feb 1994 21:52:25 -0800 (PST)
+C: From: Fred Foobar <foobar@Blurdybloop.COM>
+C: Subject: Test 04
+C: To: mooch@owatagu.siam.edu
+C: Message-Id: <B27397-0100000@Blurdybloop.COM>
+C: MIME-Version: 1.0
+C: Content-Type: TEXT/PLAIN; CHARSET=US-ASCII
+C:
+C: Test 04
```
Comment (strokyl) [inline mpt/impl/imap-mailbox/core/src/main/resources/org/apache/james/imap/scripts/ConcurrentAppendUIDToMSNMapping.test]: When you write "Test 04" does it mean the next regext group should match "04" ?

Comment (chibenwa) [inline mpt/impl/imap-mailbox/core/src/main/resources/org/apache/james/imap/scripts/ConcurrentAppendUIDToMSNMapping.test]: No. 

The definition of the regext next line should match is after **C:**

Here the regexp to match is "Test 04"

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**